app/models/pydantic/UserModel.py

```diff
@@ -32,6 +32,33 @@ def check_password(self) -> Self:
                 raise HTTPException(status_code=400, detail=CommonErrorMessages.PASSWORDS_DONT_MATCH)
         return self
 
+class PydanticUserCreate(BaseModel):
+    """
+    This model is meant to be used when we need to create a new user.
+    """
+    login: Login
+    # The admin can choose the password or let it be generated
+    password: Optional[Password] = None
+    password_confirm: Optional[Password] = None
+    firstname: Name
+    lastname: Name
+    mail: Mail
+
+
+    @model_validator(mode="after")
+    def check_password(self) -> Self:
+        """
+        This validator ensures that the two passwords provided match if they are not None.
+        This validator ensures that the two passwords are provided together or none.
+        """
+        if (self.password_confirm is None) != (self.password is None):
+            raise HTTPException(status_code=400,detail="Both 'password' and 'password_confirm' must be specified together or not at all.")
+
+        if self.password is not None:
+            if self.password != self.password_confirm:
+                raise HTTPException(status_code=400, detail=CommonErrorMessages.PASSWORDS_DONT_MATCH)
```

Comment:
The error code should be 422.
https://developer.mozilla.org/fr/docs/Web/HTTP/Status/422